packages/screens/Organizations/OrganizationDeployerScreen.tsx

```diff
@@ -1,49 +1,62 @@
 import React, { useState } from "react";
 import { StyleSheet, View } from "react-native";
+import { useSelector } from "react-redux";
 
 import { ConfigureVotingSection } from "./components/ConfigureVotingSection";
 import { CreateDAOSection } from "./components/CreateDAOSection";
 import { LaunchingOrganizationSection } from "./components/LaunchingOrganizationSection";
-import { MemberSettingsSection } from "./components/MemberSettingsSection";
+import { MemberBasedSettingsSection } from "./components/MemberBasedSettingsSection";
+import { NFTBasedSettingsSection } from "./components/NFTBasedSettingsSection";
 import { ReviewInformationSection } from "./components/ReviewInformationSection";
 import { RightSection } from "./components/RightSection";
-import { TokenSettingsSection } from "./components/TokenSettingsSection";
+import { TokenBasedSettingsSection } from "./components/TokenBasedSettingsSection";
 import {
   ConfigureVotingFormType,
   CreateDaoFormType,
+  DaoType,
   LaunchingProcessStepType,
-  TokenSettingFormType,
   MemberSettingFormType,
-  DaoType,
+  NFTSettingFormType,
+  TokenSettingFormType,
 } from "./types";
 import { BrandText } from "../../components/BrandText";
 import { ScreenContainer } from "../../components/ScreenContainer";
 import { useFeedbacks } from "../../context/FeedbacksProvider";
 import useSelectedWallet from "../../hooks/useSelectedWallet";
 import {
   NetworkKind,
-  getKeplrSigningCosmWasmClient,
   getNetwork,
   getUserId,
   mustGetCosmosNetwork,
 } from "../../networks";
-import { createDaoTokenBased, createDaoMemberBased } from "../../utils/dao";
+import { selectNFTStorageAPI } from "../../store/slices/settings";
+import {
+  createDaoTokenBased,
+  createDaoMemberBased,
+  createDaoNftBased,
+} from "../../utils/dao";
 import { adenaDeployGnoDAO } from "../../utils/gnodao/deploy";
 
 export const ORGANIZATION_DEPLOYER_STEPS = [
   "Create a DAO",
   "Configure voting",
-  "Set tokens or members",
+  "Set members or contract",
   "Review information",
   "Launch organization",
 ];
 
 export const LAUNCHING_PROCESS_STEPS: LaunchingProcessStepType[] = [
+  { title: "Book name", completeText: "Transaction finalized" },
   { title: "Create organization", completeText: "Transaction finalized" },
+  {
+    title: "Transfer name to organization",
+    completeText: "Transaction finalized",
```

Comment:
this should do a single transaction in teritori case and keep a single step

Reply:
We can only do 2 TXs minimum for now -> need contract revision